Pa3/Durchführung/Project/src/app/State/DriveToStart.c

```diff
@@ -25,27 +27,69 @@
 
 /* VARIABLES **************************************************************************************/
 static SoftTimer* gMaxTimeTimer;
+static SoftTimer* gLapTimer;
+
+LineSensorValues* sensorValues;
 
 /* EXTERNAL FUNCTIONS *****************************************************************************/
 
 void DriveToStart_StartMotorsAndTimer(void)
 {
-
+    SoftTimer_Ret status;
+
+    DriveHandler_FollowGuideLine(sensorValues);
+
+    if (SOFT_TIMER_UNREGISTERED == gMaxTimeTimer->state)
+    {
+        status = SoftTimerHandler_register(gMaxTimeTimer);
+    }
+    if (SOFTTIMER_RET_SUCCESS == status)
+    {
+        SoftTimer_start(gMaxTimeTimer, MAX_START_TIME_s);
+    }
 }
 
 EventEnum DriveToStart_FollowGuideLine(void)
 {
+    EventEnum retEvent = NO_EVENT_HAS_HAPPEND;
+
+    LineSensor_read(sensorValues);
 
+    if (SOFTTIMER_IS_EXPIRED(gMaxTimeTimer))
+    {
+        retEvent = DRIVE_TO_START_IS_ACTIVE_FOR_TOO_LONG;
+    }
+    else if ((500 < sensorValues->value[LINESENSOR_LEFT]) && (500 < sensorValues->value[LINESENSOR_RIGHT]))
+    {
+        retEvent = START_FINISH_LINE_WAS_RECOGINZED;
+    }
+
+    return retEvent;
 }
 
 void DriveToStart_StopTimer(void)
 {
-
+    SoftTimer_Stop(gMaxTimeTimer);
```

Comment:
no unregister needed?

Reply:
i do not believe unregistering the timer is strictly necessary if the number of softtimerstructures stays lower than 10 
will add unregistering the timer in order to minimize possibility of errors

Reply:
Okay, but line 42ff doesn't work then. You only set status if it is unregistered and therefore do not start the timer again. Maybe it is easier and saves resources if we just unregister it every time?